cedar-policy/src/api/err.rs

```diff
@@ -115,6 +116,197 @@ pub enum ReAuthorizeError {
     },
 }
 
+/// The kind of hierarchy that has transitive closure errors
+#[derive(Debug, Clone)]
+pub enum HierarchyKind {
+    /// An action hierarchy
+    Action,
+    /// An entity type hierarchy
+    EntityType,
+}
+
+impl std::fmt::Display for HierarchyKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Action => write!(f, "action"),
+            Self::EntityType => write!(f, "entity type"),
+        }
+    }
+}
+
+/// Transitive closure computation or enforcement error
+#[derive(Debug, Diagnostic, Error, Clone)]
+#[error("transitive closure computation/enforcement error on {kind} hierarchy: {message}")]
+pub struct TransitiveClosureError {
+    kind: HierarchyKind,
+    message: SmolStr,
+}
+
+impl TransitiveClosureError {
+    /// Getter of hierarchy kind
+    pub fn get_kind(&self) -> HierarchyKind {
```

Comment:
This should probably either return `&HierarchyKind`, or we should just make `HierarchyKind` `derive(Copy)`

Reply:
Done.